src/embedded-checkout/loading-indicator.ts

```diff
@@ -0,0 +1,120 @@
+import { LoadingIndicatorStyles } from './embedded-checkout-styles';
+
+const DEFAULT_STYLES: LoadingIndicatorStyles = {
+    size: 70,
+    color: '#d9d9d9',
+    backgroundColor: '#ffffff',
+};
+
+const ROTATION_ANIMATION = 'embedded-checkout-loading-indicator-rotation';
+
+export default class LoadingIndicator {
+    private _container: HTMLElement;
+    private _indicator: HTMLElement;
+    private _styles: LoadingIndicatorStyles;
+
+    constructor(
+        options?: { styles?: LoadingIndicatorStyles }
+    ) {
+        this._styles = { ...DEFAULT_STYLES, ...options && options.styles };
+
+        this._defineAnimation();
+
+        this._container = this._buildContainer();
+        this._indicator = this._buildIndicator();
+
+        this._container.appendChild(this._indicator);
+    }
+
+    show(parentId?: string): void {
+        if (parentId) {
+            const parent = document.getElementById(parentId);
+
+            if (!parent) {
+                throw new Error('Unable to attach the loading indicator because the parent ID is not valid.');
+            }
+
+            parent.appendChild(this._container);
+        }
+
+        if (!this._container.parentNode) {
+            throw new Error('Unable to show the loading indicator because it is not attached to the DOM.');
```

Comment:
would this only happen if `parentId` is not provided?

Reply:
Good question. I might not even need this check. I'll double check.